### PR TITLE
Passing scheduler host to start_tracker

### PR DESCRIPTION
**Pull request by mmccarty**
This was causing issues with our internal network. Please let me know if there is a more desirable fix.

### CODE REVIEW

**Comment by gforsyth** [comment]
I'm not certain we want to do this / hard-code it -- if a user is using the `dask` helm chart then the `client.scheduler.address` will resolve to the reverse-proxy service instead of the scheduler pod

**Comment by mmccarty** [comment]
Good catch! Is there any way we can test for this? I'll think of a another approach.

**Comment by mmccarty** [comment]
@gforsyth CI was flaky but looks like it is passing now. What do you think?

File: dask_xgboost/core.py [hunk 1]
```diff
@@ -35,10 +36,13 @@ def parse_host_port(address):
...
 
-def start_tracker(host, n_workers):
+def start_tracker(host, n_workers, default_host=None):
     """ Start Rabit tracker """
     if host is None:
-        host = get_host_ip("auto")
+        try:
+            host = get_host_ip("auto")
+        except socket.gaierror:
+            host = default_host
```
Comment (gforsyth) [inline dask_xgboost/core.py:45]:
```suggestion
            if default_host is not None:
                host = default_host
            else:
                raise
```

**Review by gforsyth — CHANGES_REQUESTED** [review]
Hey @mmccarty -- this looks like a reasonable approach to me, one suggestion for slightly better error handling (although errors coming from `xgboost` are always going to be cryptic)

**Comment by mmccarty** [comment]
hey @TomAugspurger I'm getting a lot of random seg faults in CI. Sorry to bug you. Do you know what's going on?

**Comment by TomAugspurger** [comment]
No idea. I've restarted the job in hopes that it goes away.

**Comment by TomAugspurger** [comment]
All green now.

**Comment by mmccarty** [comment]
Thanks!

**Pull request merged**